Mathlib/Algebra/Homology/Embedding/RestrictionHomology.lean

```diff
@@ -48,4 +48,125 @@ lemma hasHomology [K.HasHomology j'] : (K.restriction e).HasHomology j :=
 
 end restriction
 
+variable (i j k : ι) (hi : c.prev j = i) (hk : c.next j = k)
+  {i' j' k' : ι'} (hi' : e.f i = i') (hj' : e.f j = j') (hk' : e.f k = k')
+  (hi'' : c'.prev j' = i') (hk'' : c'.next j' = k')
```

Comment:
We have `(e : c.Embedding c') [e.IsRelIff]`, do we also need all of these hypotheses? I kind of feel that by this point something should imply something, e.g. do `hi` and `hi'` and `hj'` imply `hi''`? Or is the point that even if this is true, we should have these hypotheses to hand anyway becuase they are more convenient?

I would have chosen notation `hji` for `hi`, `hjk` for `hk`, `hji'` for `hi''` and `hjk'` for `hk''` but this is a minor point.

Reply:
The equality `c.prev j = i` does not necessarily imply that `i` and `j` are related by the relation `c.Rel` (this happens if there is no nonzero differential towards degree `j`). Then, I do not think the assumptions are redundant!